core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala

```diff
@@ -51,7 +51,26 @@ private case class ExecutorRegistered(executorId: String)
 
 private case class ExecutorRemoved(executorId: String)
 
-private[spark] case class HeartbeatResponse(reregisterBlockManager: Boolean)
+private[spark] case class UpdatedEpoch(epoch: Long)
+
+private[spark] object HeartbeatResponse {
+  def apply(reregisterBlockManager: Boolean,
+            updatedEpoch: Option[Long] = None): HeartbeatResponse =
+    updatedEpoch.fold[HeartbeatResponse](BasicHeartbeatResponse(reregisterBlockManager)) {
```

Comment:
nit: a while back I think the community decided that `Option.fold` is pretty confusing for folks that aren't used to scala, with little gain, so I'd get rid of it
(though given my other comment about the unnecessary class, you can probably just remove this whole thing.)

Reply:
Makes sense, this code should be adapted to follow the code to the Spark code base conventions and style